iot/api-client/mqtt_example/src/main/java/com/google/cloud/iot/examples/MqttExample.java

```diff
@@ -123,22 +122,29 @@ public static void main(String[] args) throws Exception {
     MqttClient client = new MqttClient(mqttServerAddress, mqttClientId, new MemoryPersistence());
     client.connect(connectOptions);
 
+    // Publish to the events or state topic based on the flag.
+    String subTopic = options.messageType.equals("event") ? "events" : options.messageType;
+
     // The MQTT topic that this device will publish telemetry data to. The MQTT topic name is
     // required to be in the format below. Note that this is not the same as the device registry's
     // Cloud Pub/Sub topic.
-    String mqttTopic = String.format("/devices/%s/events", options.deviceId);
+    String mqttTopic = String.format("/devices/%s/%s", options.deviceId, subTopic);
 
     // Publish numMessages messages to the MQTT bridge, at a rate of 1 per second.
     for (int i = 1; i <= options.numMessages; ++i) {
       String payload = String.format("%s/%s-payload-%d", options.registryId, options.deviceId, i);
-      System.out.format("Publishing message %d/%d: '%s'\n", i, options.numMessages, payload);
+      System.out.format(
+          "Publishing %s message %d/%d: '%s'\n",
+          options.messageType, i, options.numMessages, payload);
 
       // Publish "payload" to the MQTT topic. qos=1 means at least once delivery. Cloud IoT Core
       // also supports qos=0 for at most once delivery.
       MqttMessage message = new MqttMessage(payload.getBytes());
       message.setQos(1);
       client.publish(mqttTopic, message);
-      Thread.sleep(1000);
+
+      // Send events every second. State should not be updated as often
+      Thread.sleep(options.messageType.equals("event") ? 1000 : 5000);
     }
     // Disconnect the client and finish the run.
     client.disconnect();
```

Comment:
is this automatically done if there is an exception ? Is it worth moving this to a `finally` block ?